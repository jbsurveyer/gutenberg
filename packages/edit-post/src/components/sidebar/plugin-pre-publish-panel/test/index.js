/**
 * External dependencies
 */
import ReactTestRenderer from 'react-test-renderer';

/**
 * WordPress dependencies
 */
import { SlotFillProvider } from '@wordpress/components';

/**
 * Internal dependencies
 */
import PluginPrePublishPanel from '../';

jest.mock( '../../../../../../components/src/button' );

describe( 'PluginPrePublishPanel', () => {
	test( 'renders fill properly', () => {
		const tree = ReactTestRenderer.create(
			<SlotFillProvider>
				<PluginPrePublishPanel
					className="my-plugin-pre-publish-panel"
					title="My panel title"
					initialOpen={ true }
				>
					My panel content
				</PluginPrePublishPanel>
				<PluginPrePublishPanel.Slot />
			</SlotFillProvider>
		).toJSON();

		expect( tree ).toMatchSnapshot();
	} );
} );
