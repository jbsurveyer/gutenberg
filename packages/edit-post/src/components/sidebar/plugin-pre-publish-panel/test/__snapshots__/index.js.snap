// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PluginPrePublishPanel renders fill properly 1`] = `
<div
  className="components-panel__body my-plugin-pre-publish-panel is-opened"
>
  <h2
    className="components-panel__body-title"
  >
    <button
      aria-expanded={true}
      className="components-button components-panel__body-toggle"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="components-panel__arrow"
        focusable="false"
        height="24px"
        role="img"
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M0,0h24v24H0V0z"
            fill="none"
          />
        </g>
        <g>
          <path
            d="M12,8l-6,6l1.41,1.41L12,10.83l4.59,4.58L18,14L12,8z"
          />
        </g>
      </svg>
      My panel title
    </button>
  </h2>
  My panel content
</div>
`;
